charts/external-dns/templates/deployment.yaml

```diff
@@ -132,6 +136,47 @@ spec:
           resources:
             {{- toYaml . | nindent 12 }}
           {{- end }}
+        {{- if eq $provider "webhook" }}
+        {{- with .Values.provider }}
+        - name: {{ $providername }}
+          image: {{ get $whp .name }}
+          imagePullPolicy: {{ $.Values.image.pullPolicy }}
+          {{- with .env }}
+          env:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          {{- with .args }}
+          args:
+            - {{ toYaml . | nindent 12 }}
+          {{- end }}
+          ports:
+            - name: http
+              protocol: TCP
+              containerPort: 8888
+          livenessProbe:
+            {{- toYaml $.Values.livenessProbe | nindent 12 }}
+          readinessProbe:
+            {{- toYaml $.Values.readinessProbe | nindent 12 }}
+          {{- if or $.Values.secretConfiguration.enabled $.Values.extraVolumeMounts }}
+          volumeMounts:
+            {{- if $.Values.secretConfiguration.enabled }}
+            - name: secrets
+              mountPath: {{ tpl $.Values.secretConfiguration.mountPath $ }}
+            {{- with $.Values.secretConfiguration.subPath }}
+              subPath: {{ tpl . $ }}
+            {{- end }}
+            {{- end }}
+            {{- with $.Values.extraVolumeMounts }}
```

Comment:
Likewise, extra volume mounts should not be shared across the main container and sidecar.

Reply:
Same here: Wdyt then about mounting `extraVolumes` only on external-dns for in-tree provider and only on the sidecar for webhook providers ?
cc @stevehipwell @mrueg

Reply:
I would think that `provider` should have it's own `extraVolumeMounts` so it can be treated seperatly.

Reply:
Implemented with 91095975